### ISSUE TITLE
Option in the Yeoman Generator to generate a "blank" project (without "hello world")

### ISSUE THREAD

**Issue by ebousse** [eclipse-langium/langium#1764]
## Problem

The Langium Yeoman generator is great to kickstart a working project around a small "hello world" DSL, and the provided examples of code to be written around the AST (eg. the example validator) are very useful to understand Langium.

However, a typical flow is:

1. generate a langium project with `yo langium`
2. modify the default `.langium` file,
3. run `npm run langium:generate`
4. run `npm run build`

If no other change is done, **step 4 will always fail** because the provided examples of generator, validator and CLI won't compile anymore. This is because these examples are made for the "hello world" default DSL, which is not there anymore because of step 2.

This means that two extra steps are in fact always required:

5. modify generator, validator and CLI  by commenting/changing/removing the parts that cannot compile
6. run `npm run build` once again, and hope it compiles

A problem I encountered when I was starting with Langium, and that I now currently encounter when I try to teach Langium to students, is that the additional **step 5 is quite difficult for people who are just starting with Langium**. Because at this stage one does not know how things works, it is hard to know how to make the changes, and it is easy to make mistakes (eg. commenting too many things).

## Idea

What I would find very useful would be to be offered the choice, in the Yeoman generator, to either have a project with the "hello world" example DSL (and accompanying generator, validator and CLI), or to have a "blank" project without any semantic model nor grammar rules. Such blank project could still provide files for the generator, validator and CLI, but where one would simply find `// TODO`  in each place where a change will be required for the component to work in the future.

This would allow kickstarting a project with steps 1 to 4, avoiding the hazardous steps 5-6 (especially for beginners).

**Comment by msujew** [comment]
I agree, this makes a lot of sense. I've talked with a few people at conferences in the past who had this issue as well.

**Comment by Mistourr** [comment]
Hello, as part of a software engineering course me and my classmates would really be willing to help solve this issue. However we're left wondering what would the Langium team prefers between : 
- Having two different templates ( possibly more maintenance )
- Use of a templating engine ( new dependency )

On top of that we still wonder if the minimal example should include comments for guidelines and if it should compile ( meaning it still needs a minimum grammar and some methods that will need changing )

**Comment by spoenemann** [comment]
My view: 
- The CLI doesn't make sense without a concrete example, so I'd disable it if a blank project is desired
- The extension is not affected (contains only generic code)
- In the language package, the grammar file and validator are affected; for those two, I'd simply create different templates
- The tests don't make sense without an example, so I'd skip those
